### ISSUE TITLE
Issue with GRU Based Autoencoder

### ISSUE THREAD

**Issue by AshimaChawla** [keras-team/keras#12422]
I am trying to train GRU based Autoencoder but the inference model for decoder doesn't seem to work.

def define_models(n_input, n_output, n_units):
    #n_units= latent dimension 
    # define training encoder
    encoder_inputs = Input(shape=(None, n_input))
    encoder = GRU(n_units, return_state=True)
    encoder_outputs, state_h = encoder(encoder_inputs)
    encoder_states = [state_h]
     
    # define training decoder
    decoder_inputs = Input(shape=(None, n_output))
    decoder_gru = GRU(n_units, return_sequences=True, return_state=True)
    decoder_outputs = decoder_gru(decoder_inputs, initial_state=state_h)
    decoder_dense = Dense(n_output, activation='softmax')
    decoder_outputs = decoder_dense(decoder_outputs)
    model = Model([encoder_inputs, decoder_inputs], decoder_outputs)
     
     
    # define inference encoder
    encoder_model = Model(encoder_inputs, encoder_states)
     
     
    # define inference decoder
    decoder_state_input_h = Input(shape=(n_units,))
    decoder_states_inputs = [decoder_state_input_h]
    decoder_output, state_h = decoder_gru(decoder_inputs, initial_state=decoder_states_inputs)
    decoder_states = [state_h]
    decoder_outputs = decoder_dense(decoder_output)
    decoder_model = Model([decoder_inputs] + decoder_states_inputs, [decoder_outputs] + decoder_states)
     
     
    # return all models
    return model, encoder_model, decoder_model
